api/v1alpha1/randomsecret_types.go

```diff
@@ -306,7 +307,11 @@ func (r *RandomSecret) validateEitherPasswordPolicyReferenceOrInline() error {
 func (r *RandomSecret) validateInlinePasswordPolicyFormat() error {
 	if r.Spec.SecretFormat.InlinePasswordPolicy != "" {
 		passwordPolicyFormat := &PasswordPolicyFormat{}
-		return hclsimple.Decode("dummy.hcl", []byte(r.Spec.SecretFormat.InlinePasswordPolicy), nil, passwordPolicyFormat)
+		if strings.HasSuffix(".hcl", r.Spec.SecretKey) {
```

Comment:
The `HasSuffix` function requires the variable being tested to be the first parameter. Swap the parameters